apps/portal/app/components/save-list/save-list-modal.tsx

```diff
@@ -77,117 +73,71 @@ export default function SaveListModal({
+    if (vaultDetails) {
+      setIsLoading(false)
+    }
+  }, [vaultDetails])
 
   const {
-    writeContractAsync,
-    receipt: txReceipt,
+    mutateAsync: stake,
+    txReceipt,
     awaitingWalletConfirmation,
     awaitingOnChainConfirmation,
     isError,
     reset,
-  } = mode === 'save' ? depositHook : redeemHook
-
-  const [vaultDetails, setVaultDetails] = useState<VaultDetailsType>()
-  const vaultDetailsFetcher = useFetcher<VaultDetailsType>()
-
-  const { id: vaultId } = useAtomValue(saveListModalAtom)
-
-  useEffect(() => {
-    let isCancelled = false
+  } = useSaveListMutation(
+    contract,
+    vaultDetails?.user_conviction ?? '0',
+    mode as 'deposit' | 'redeem',
+  )
 
-    if (vaultId !== null) {
-      const finalUrl = `${GET_VAULT_DETAILS_RESOURCE_ROUTE}?contract=${contract}&vaultId=${vaultId}&fetchId=${fetchId}`
-      if (!isCancelled) {
-        vaultDetailsFetcher.load(finalUrl)
-      }
-    }
+  const handleAction = async () => {
+    try {
+      const txHash = await stake({
+        val,
+        userWallet,
+        vaultId: vaultId ?? '',
+      })
 
-    return () => {
-      isCancelled = true
-    }
-    // omits the fetcher from the exhaustive deps
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [contract, vaultId, fetchId])
+      if (publicClient && txHash) {
+        dispatch({ type: 'TRANSACTION_PENDING' })
+        const receipt = await publicClient.waitForTransactionReceipt({
+          hash: txHash,
+        })
 
-  useEffect(() => {
-    if (vaultDetailsFetcher.state === 'loading') {
-      setIsLoading(true)
-    }
-  }, [vaultDetailsFetcher.state])
+        dispatch({
+          type: 'TRANSACTION_COMPLETE',
+          txHash,
+          txReceipt: receipt,
+        })
 
-  useEffect(() => {
-    if (vaultDetailsFetcher.state === 'idle' && vaultDetailsFetcher.data) {
-      setVaultDetails(vaultDetailsFetcher.data)
-      setIsLoading(false)
-    }
-  }, [vaultDetailsFetcher.state, vaultDetailsFetcher.data])
-
-  const useHandleAction = (actionType: string) => {
-    return async () => {
-      try {
-        if (!contract || !vaultId || !vaultDetails) {
-          throw new Error('Missing required parameters')
-        }
-        const txHash = await writeContractAsync({
-          address: contract as `0x${string}`,
-          abi: multivaultAbi as Abi,
-          functionName:
-            actionType === 'save' ? 'depositTriple' : 'redeemTriple',
-          args:
-            actionType === 'save'
-              ? [userWallet as `0x${string}`, vaultId]
-              : [
-                  vaultDetails.user_conviction,
-                  userWallet as `0x${string}`,
-                  vaultId,
-                ],
-          value:
-            actionType === 'save'
-              ? parseUnits(val === '' ? '0' : val, 18)
-              : undefined,
+        await queryClient.refetchQueries({
+          queryKey: ['get-vault-details', contract, vaultId],
```

Comment:
We may need to also invalidate the `get-account-details` query but mainly making a note about this now -- we can check when we're finetuning our revalidations